Allow disabling notifications per user and disabling specific notifications per user

Kukks opened the pull request:
closes #1974

Someone can prettify this UI afterwards. The reason that you disable instead of enable specific notifications is because if we add new notifications, users would not get them by default.

![image](https://user-images.githubusercontent.com/1818366/96228884-75c09b00-0f96-11eb-9a9b-f48793555432.png)

pavlenex commented:
Nice. Concept ACK, but the UI needs some checkboxes love, (like we have for policies, notifications, etc) if @dennisreimann is willing to halp and improve it ✅

NicolasDorier commented:
I would also really prefer checkboxes.

britttttk commented:
@Kukks what is Junk in this list for?

Kukks commented:
> @Kukks what is Junk in this list for?

It's a way to test notifications when in Debug mode

britttttk commented:
Will the checkboxes fix this or it's a bug? I'm not sure
![image](https://user-images.githubusercontent.com/39231115/96331968-a3512700-101e-11eb-81ff-8dd85574bfbd.png)

Kukks commented:
> Will the checkboxes fix this or it's a bug? I'm not sure
> ![image](https://user-images.githubusercontent.com/39231115/96331968-a3512700-101e-11eb-81ff-8dd85574bfbd.png)

That was a bug, fixed now. Thanks!

Kukks commented:
![image](https://user-images.githubusercontent.com/1818366/96333560-fcdf4300-106a-11eb-9eb8-1ab51291b86e.png)


here you go

britttttk commented:
tACK - They are all working now (note: I didn't test version disable since it already existed in admin policy).

pavlenex commented:
@Kukks Can `Junk` be renamed? It's not really clear what it does, renaming it to something more sensible would really be helpful.
As Brit mentioned, since we do have "new update" in Server Settings > Policies, I don't think we really need them for user-settings?

Kukks commented:
It doesn't matter, it doesn't get included in releases at all.

On Sun, Oct 18, 2020 at 9:13 AM Pavlenex <notifications@github.com> wrote:

> @Kukks <https://github.com/Kukks> Can Junk be renamed? It's not really
> clear what it does, renaming it to something more sensible would really be
> helpful.
> As Brit mentioned, since we do have "new update" in Server Settings >
> Policies, I don't think we really need them for user-settings?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/btcpayserver/btcpayserver/pull/1991#issuecomment-711127527>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAN357U6KTCPAWSSED3CH2LSLKISLANCNFSM4STAACQA>
> .
>

Kukks commented:
> As Brit mentioned, since we do have "new update" in Server Settings > Policies, I don't think we really need them for user-settings?

The two things are unrelated tech wise. One issues version notifications, the other is a user rejecting those types of notifications.

It makes sense to have both as you can have multiple admins where >=1 admins would still want to know about it

dennisreimann reviewed (commented):
Would it be an option to extract this inline script to a separate file, which could be included in the main bundle?

dennisreimann commented on BTCPayServer/Views/Manage/NotificationSettings.cshtml:
```diff
@@ -0,0 +1,51 @@
+        <div class="form-group">
+            <label asp-for="DisabledNotifications">Do not receive notifications for</label>
+
+            <select size="@items.Count()" class="form-control hide-when-js" multiple asp-for="DisabledNotifications"
```
Tdchnically/HTML-wise: Why is this a multiselect instead of regular checkboxes?

Kukks replied:
because the data model is a string array vs an object with `id, selected` props

dennisreimann replied:
Ok, but we could transform it for the view, right? 
I'd find this to be cleaner than having different approaches for JS/Non-JS, plus we'd get rid of the extra bootstrap-multiselect.js

Kukks replied:
done

Kukks commented:
> Would it be an option to extract this inline script to a separate file, which could be included in the main bundle?

Yes, but I would wait until we need to reuse it elsewhere

pavlenex commented:
We can re-use this in Server Settings > Policies perhaps?

dennisreimann commented:
> Yes, but I would wait until we need to reuse it elsewhere

My concern is more to save those extra bytes per each request where the snippet is included. Currently it's part of each request instead of once in the bundle. Yes, it's not much, but we could optimize it.

Kukks commented:
> > Yes, but I would wait until we need to reuse it elsewhere
> 
> My concern is more to save those extra bytes per each request where the snippet is included. Currently it's part of each request instead of once in the bundle. Yes, it's not much, but we could optimize it.

Removed the js dependency, it was definitely a lazy move from my end when i added it in
